sdcm/provision/azure/virtual_machine_provider.py

```diff
@@ -80,6 +84,19 @@ def get_or_create(self, definitions: List[InstanceDefinition], nics_ids: List[st
                     }],
                 },
             }
+
+            if self._enable_azure_kms:
+                self._kms_provider = AzureKmsProvider(
+                    self._resource_group_name, self._region, self._az, self._azure_service)
+                # Extract test_id from resource group name
+                test_id = self._resource_group_name.split(SCT_RESOURCE_GROUP_PREFIX)[-1][:36]
```

Comment:
This approach looks as unneeded implicit and unsafe way for getting what we have already in other attributes.
Both, `tester` and `test_config` modules have the `test_id` attr.

Reply:
> Ok, updated PR to use TestConfig for get test_id.

This caused circular dependency, reverted the change. Will check whether `tester` can be used.